src/examples/dropdowns/combobox/Default.tsx

```diff
@@ -0,0 +1,103 @@
+/**
+ * Copyright Zendesk, Inc.
+ *
+ * Use of this source code is governed under the Apache License, Version 2.0
+ * found at http://www.apache.org/licenses/LICENSE-2.0.
+ */
+
+import React, { useMemo, useState, useEffect } from 'react';
+import debounce from 'lodash.debounce';
+import { Item, Menu, Label, Field, Dropdown, Combobox } from '@zendeskgarden/react-dropdowns';
+import { Row, Col } from '@zendeskgarden/react-grid';
+import { ReactComponent as SearchIcon } from '@zendeskgarden/svg-icons/src/16/search-stroke.svg';
+import { Dots } from '@zendeskgarden/react-loaders';
+
+const options = [
+  'Asparagus',
+  'Brussel sprouts',
+  'Cauliflower',
+  'Garlic',
+  'Jerusalem artichoke',
+  'Kale',
+  'Lettuce',
+  'Onion',
+  'Mushroom',
+  'Potato',
+  'Radish',
+  'Spinach',
+  'Tomato',
+  'Yam',
+  'Zucchini'
+];
+
+const Example = () => {
+  const [isLoading, setLoading] = useState(false);
+  const [selectedItem, setSelectedItem] = useState('');
+  const [inputValue, setInputValue] = useState('');
+  const [matchingOptions, setMatchingOptions] = useState<string[]>([]);
+
+  /**
+   * Debounce filtering
+   */
+  const filterMatchingOptions = useMemo(
+    () =>
+      debounce((value: string) => {
+        if (value.length > 0) {
+          setMatchingOptions(
+            options.filter(
+              option => option.trim().toLowerCase().indexOf(value.trim().toLowerCase()) !== -1
+            ) || []
+          );
+        } else {
+          setMatchingOptions([]);
+        }
+        setLoading(false);
+      }, 750),
+    [setMatchingOptions]
```

Comment:
Alternatively, can remove `setMatchingOptions` from dependency array since it's a set state from React and will not be changing each render.

Reply:
You are right, will update and remove.